### ISSUE TITLE
`R CellCensus package checks` CI workflow is slow

### ISSUE THREAD

**Issue by bkmartinjr** [chanzuckerberg/cellxgene-census#309]
The GHA workflow `R CellCensus package checks` is very slow - taking 30+ minutes to complete.  It would be wonderful if this could be optimized at some point.

**Comment by mlin** [comment]
Agreed -- this will improve once R tiledbsoma has stabilized enough that we can depend on a specified version number of it. Right now we don't cache dependency installation (which is particularly slow in R) so that we always get the "bleeding edge" build with the very latest patches.

**Comment by brianraymor** [comment]
@ebezzi to discuss with @mlin to understand status.

**Comment by mlin** [comment]
We're currently still tracking the bleeding edge of `tiledbsoma`, which leads to frequent CI breakage; discussion on how to ensure a stable dependency started: https://github.com/single-cell-data/TileDB-SOMA/issues/1412#issuecomment-1562368924

**Comment by bkmartinjr** [comment]
@mlin - is this still an issue or can we close this as "good enough for now"?

**Comment by mlin** [comment]
@brianraymor It's still the case that [this build is often slow](https://github.com/chanzuckerberg/cellxgene-census/actions/workflows/r-check.yml?query=branch%3Amain), although the specific low-hanging fruit [mentioned above](https://github.com/chanzuckerberg/cellxgene-census/issues/309#issuecomment-1572647627) was picked.

A further improvement we could make would be to add some customized caching logic to the [`tiledbsoma_latest` build](https://github.com/chanzuckerberg/cellxgene-census/blob/acfde7100936eef16b317b708997372dd3045a20/.github/workflows/r-check.yml#L18) OR move it into some non-blocking periodic (cron) job. That build [installs R tiledbsoma directly from git](https://github.com/chanzuckerberg/cellxgene-census/blob/acfde7100936eef16b317b708997372dd3045a20/.github/workflows/r-check.yml#L35), which takes time and is currently never cached (but could be in principle, if that main branch hasn't changed).

We could also look at adding more stuff to [paths-ignore](https://github.com/chanzuckerberg/cellxgene-census/blob/acfde7100936eef16b317b708997372dd3045a20/.github/workflows/r-check.yml#L5) so that this build is simply not triggered as much for diffs that didn't actually change our R code (which has changed little recently).

I think it's fine either to close this issue or leave open for a future time if/when R package development is more prioritized, defer to you.
